lib/Helper/ParameterBagHelper.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This source file is available under the terms of the
+ * Pimcore Open Core License (POCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (https://www.pimcore.com)
+ *  @license    Pimcore Open Core License (POCL)
+ */
+
+namespace Pimcore\Helper;
+
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+/**
+ * Helper class for safely extracting typed values from ParameterBag instances.
+ *
+ * This helper addresses Symfony 7 breaking changes where ParameterBag::getInt()
+ * and ParameterBag::getBool() now throw UnexpectedValueException instead of
+ * returning fallback values. This helper uses the filter() method internally
+ * with FILTER_NULL_ON_FAILURE to maintain backward compatibility.
+ *
+ * @internal
+ */
+final class ParameterBagHelper
+{
+    /**
+     * Safely retrieves an integer value from a ParameterBag.
+     *
+     * Uses FILTER_VALIDATE_INT with FILTER_NULL_ON_FAILURE to avoid exceptions.
+     * Returns the default value if the key doesn't exist or contains an invalid value.
+     *
+     * Examples:
+     * // Basic usage
+     * $id = ParameterBagHelper::getInt($request->query, 'id');
+     *
+     * // With default value
+     * $limit = ParameterBagHelper::getInt($request->request, 'limit', 50);
+     *
+     * // Multiple sources with fallback
+     * $id = ParameterBagHelper::getInt($request->attributes, 'id')
+     *     ?: ParameterBagHelper::getInt($request->query, 'id');
+     *
+     */
+    public static function getInt(ParameterBag $bag, string $key, int $default = 0): int
+    {
+        $env = $_SERVER['APP_ENV'] ?? 'prod';
+        if (\in_array($env, ['dev', 'staging'], true)) {
+            $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2);
+            $caller = $backtrace[1] ?? [];
+            $callerInfo = isset($caller['class'], $caller['function'])
+                ? sprintf('%s::%s()', $caller['class'], $caller['function'])
+                : ($caller['function'] ?? 'unknown');
+
+            trigger_deprecation(
+                'pimcore/pimcore',
+                '12.3',
+                sprintf('Usage of getInt is deprecated. Use proper parameter validation instead in %s.', $callerInfo)
+            );
```

Comment:
No need for `sprintf` here, `trigger_deprecation` has it built in:

```suggestion
            trigger_deprecation(
                'pimcore/pimcore',
                '12.3',
                'Usage of %s() is deprecated. Use proper parameter validation instead in %s.',
                __METHOD__,
                $callerInfo,
            );
```

Reply:
@jdreesen Good catch, thanks! 😊
Didn’t realize trigger_deprecation() already supports sprintf-style placeholders — makes it cleaner indeed.
I’ll update it accordingly 👍